### PR TITLE
config: add 'C_' prefix

### PR DESCRIPTION
**Pull request by flatcap**
Following POLL #1572 
All the Config variables are now uniquely named.
If anyone has a better suggestion for the naming, it's easily done.

There are no functional changes.
I'm not expecting anyone to read the entire diff :-)

---

## Rename **Config** global variables

There are over 400 config variables in NeoMutt, each of them backed by a global variable in the code.
The global variable's name is the "MixedCase" transform of the "snake_case" config variable.

But... Many of the config variables have short names which make them hard to seach for, e.g.

- Copy
- From
- Help
- Move
- Sort

### Suggestion

Add a prefix to all **Config** global variables, e.g.

- C_Copy
- C_From
- C_Help
- C_Move
- C_Sort

### CODE REVIEW

File: browser.c [hunk 1]
```diff
@@ -1100,7 +1100,7 @@ void mutt_select_file(char *file, size_t filelen, int flags, char ***files, int
   bool mailbox = (flags & MUTT_SEL_MAILBOX);
 
   /* Keeps in memory the directory we were in when hitting '='
-   * to go directly to $folder (Folder)
+   * to go directly to $folder (#C_Folder)
```
Comment (Austin-Ray) [inline browser.c]: Is the inclusion of the `#` intentional?

File: compose.c [hunk 2]
```diff
@@ -85,9 +85,9 @@
 #endif
 
 /* These Config Variables are only used in compose.c */
-char *ComposeFormat; ///< Config: printf-like format string for the Compose panel's status bar
-char *Ispell;           ///< Config: External command to perform spell-checking
-unsigned char Postpone; ///< Config: Save messages to the Postponed folder
+char *C_ComposeFormat; ///< Config: printf-like format string for the Compose panel's status bar
+char *C_Ispell; ///< Config: External command to perform spell-checking
+unsigned char C_Postpone; ///< Config: Save messages to the #C_Postponed folder
```
Comment (Austin-Ray) [inline compose.c]: Is the `#` intentional?

File: config/mbtable.h [hunk 3]
```diff
@@ -28,8 +28,8 @@ struct ConfigSet;
 /**
  * struct MbTable - multibyte character table
  *
- * Allows for direct access to the individual multibyte characters in a
- * string.  This is used for the Flagchars, Fromchars, StatusChars and Tochars
+ * Allows for direct access to the individual multibyte characters in a string.
+ * This is used for the #C_FlagChars, #C_FromChars, #C_StatusChars and #C_ToChars
```
Comment (Austin-Ray) [inline config/mbtable.h]: Is the `#` intentional?

File: conn/ssl.c [hunk 4]
```diff
@@ -130,7 +130,7 @@ struct SslSockData
  * loaded into the trusted store.  This function filters out expired certs.
  *
  * Previously the code used this form:
- *     SSL_CTX_load_verify_locations (ssldata->ctx, CertificateFile, NULL);
+ *     SSL_CTX_load_verify_locations (ssldata->ctx, #C_CertificateFile, NULL);
```
Comment (Austin-Ray) [inline conn/ssl.c]: Is the `#` intentional?

File: conn/ssl_gnutls.c [hunk 5]
```diff
@@ -241,7 +241,7 @@ static int tls_check_stored_hostname(const gnutls_datum_t *cert, const char *hos
 }
 
 /**
- * tls_compare_certificates - Compare certificates against CertificateFile
+ * tls_compare_certificates - Compare certificates against #C_CertificateFile
```
Comment (Austin-Ray) [inline conn/ssl_gnutls.c]: Is the `#` intentional?

File: copy.c [hunk 6]
```diff
@@ -348,7 +348,8 @@ int mutt_copy_hdr(FILE *in, FILE *out, LOFF_T off_start, LOFF_T off_end,
       if (flags & (CH_DECODE | CH_PREFIX))
       {
         if (mutt_write_one_header(out, 0, headers[x], (flags & CH_PREFIX) ? prefix : 0,
```
Comment (Austin-Ray) [inline copy.c]: This is unrelated to the naming, but this `if` statement should be modified to be more read able. Moving `(flags & CH_PREFIX) ? prefix : 0` and `mutt_window_wrap_cols(...)` out will make it significantly easier to understand.

File: copy.h [hunk 7]
```diff
@@ -49,7 +49,7 @@ struct Mailbox;
 #define CH_DECODE         (1 << 2)  /**< do RFC1522 decoding? */
 #define CH_XMIT           (1 << 3)  /**< transmitting this message? */
 #define CH_FROM           (1 << 4)  /**< retain the "From " message separator? */
-#define CH_PREFIX         (1 << 5)  /**< use IndentString string? */
+#define CH_PREFIX         (1 << 5)  /**< use #C_IndentString string? */
```
Comment (Austin-Ray) [inline copy.h]: Is the `#` intentional?

File: edit.c [hunk 8]
```diff
@@ -86,7 +86,7 @@ static char *EditorHelp2 =
  * @param[out] buflen Bytes of buffer used
  * @param[in]  offset Start reading at this file offset
  * @param[in]  bytes  Read this many bytes
- * @param[in]  prefix If true, prefix the lines with the IndentString
+ * @param[in]  prefix If true, prefix the lines with the #C_IndentString
```
Comment (Austin-Ray) [inline edit.c]: Is the `#` intentional?

**Comment by Austin-Ray** [comment]
Can you modify your commit message to explain _why_ we're prefixing with `C_`? Other than that, everything looks good.

File: copy.c (hunk 6 above)
Comment (flatcap) [reply copy.c]: Yes, definitely.  There's a lot of code like this.
I guess the condition started more simply and evolved.

I've made a note.  I usually keep a `trivial` branch on the neomutt repo to collect these important little fixes.  Feel free to create one and dump stuff in it.

**Comment by flatcap** [comment]
> Is the `#` intentional?

Yes.  Doxygen will auto-link a lot of symbols in comments.

Bare struct names, `Context` and functions, `mutt_pager()` will be picked up without any other annotation.  Globals, #defines and enums need a `#` prefix.

Note: this only applies to Doxygen comment blocks:
- `/** ... */`
- `/**< ... */`
- `///< ...`

> Can you modify your commit message to explain _why_ we're prefixing with `C_`?

Yes, will do.

File: copy.c (hunk 6 above)
Comment (Austin-Ray) [reply copy.c]: Duly noted.

**Comment by Austin-Ray** [comment]
> Yes. Doxygen will auto-link a lot of symbols in comments.

Today I learned

> I'm not expecting anyone to read the entire diff :-)

I'll have you know I read the entire diff because I did not read that line.

**Pull request merged**